pkg/objects/io.go

```diff
@@ -101,25 +95,24 @@ func ReaderToSeeker(r io.Reader) (io.ReadSeeker, error) {
 // the passed io object will be seeked to its beginning and will seek back to the
 // beginning after reading its content.
 func ComputeChecksum(data io.ReadSeeker) (string, error) {
-	// #nosec: G501
-	hash := md5.New() //nolint:gosec  // MD5 is used for checksums, not for hashing passwords
+	hash := sha256.New()
 	if _, err := io.Copy(hash, data); err != nil {
 		return "", fmt.Errorf("could not read file: %w", err)
 	}
 
 	if _, err := data.Seek(0, io.SeekStart); err != nil { // seek back to beginning of file
-		return "", fmt.Errorf("%s: %w", seekError, err)
+		return "", fmt.Errorf("%s: %w", ErrSeekError, err)
 	}
 
-	return base64.StdEncoding.EncodeToString(hash.Sum(nil)), nil
+	return fmt.Sprintf("%x", hash.Sum(nil)), nil
```

Comment:
The tests for this function are failing due to the md5 -> sha256 change and need to be updated.